src/interpreter/interpreter-non-det.ts

```diff
@@ -552,7 +559,12 @@ export const evaluators: { [nodeType: string]: Evaluator<es.Node> } = {
   },
 
   IfStatement: function*(node: es.IfStatement | es.ConditionalExpression, context: Context) {
```

Comment:
Would it be better to extract the code within `IfStatement` to another function , `evaluateConditional`,  - that both `IfStatement` and `ConditionalExpression` could call?  <br /> So that `IfStatement` would only have to support `es.IfStatement`.

Reply:
Yes. This makes sense

Reply:
Done